components/homme/cmake/machineFiles/anvil-bfb.cmake

```diff
@@ -0,0 +1,66 @@
+# CMake initial cache file for Anvil
+#
+# 2017/11 MT:  Set NETCDF and PNETCDF paths using same approach as CIME
+#              Rely on CIME's FindNetCDF and FindPnetCDF for all remaining config
+#
+SET (CMAKE_Fortran_COMPILER mpif90 CACHE FILEPATH "")
+SET (CMAKE_C_COMPILER mpicc CACHE FILEPATH "")
+SET (CMAKE_CXX_COMPILER mpicxx CACHE FILEPATH "")
+
+SET (WITH_PNETCDF FALSE CACHE FILEPATH "")
+#
+# anvil module system doesn't set environment variables, but will put
+# nc-config in our path.  anvil seperates C and Fortran libraries,
+# so we need to set both paths instead of NETCDF_DIR
+#
+EXECUTE_PROCESS(COMMAND nf-config --prefix
+  RESULT_VARIABLE NFCONFIG_RESULT
+  OUTPUT_VARIABLE NFCONFIG_OUTPUT
+  ERROR_VARIABLE  NFCONFIG_ERROR
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+SET (NetCDF_Fortran_PATH "${NFCONFIG_OUTPUT}" CACHE STRING "")
+
+EXECUTE_PROCESS(COMMAND nc-config --prefix
+  RESULT_VARIABLE NCCONFIG_RESULT
+  OUTPUT_VARIABLE NCCONFIG_OUTPUT
+  ERROR_VARIABLE  NCCONFIG_ERROR
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+SET (NetCDF_C_PATH "${NCCONFIG_OUTPUT}" CACHE STRING "")
+
+EXECUTE_PROCESS(COMMAND mpif90 --version
+  RESULT_VARIABLE CPR_RESULT
+  OUTPUT_VARIABLE CPR_OUTPUT
+  ERROR_VARIABLE  CPR_ERROR
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+IF ("${CPR_OUTPUT}" MATCHES "ifort.*")
+  SET (HOMME_USE_MKL "TRUE" CACHE FILEPATH "") # for Intel
+  #turn on preqxx target and thus strict fpmodel for F vs CXX comparison
+  SET (ADD_Fortran_FLAGS "-traceback -fp-model strict -qopenmp -O1" CACHE STRING "")
+  SET (ADD_C_FLAGS "-traceback -fp-model strict -qopenmp -O3" CACHE STRING "")
+  SET (ADD_CXX_FLAGS "-traceback -fp-model strict -qopenmp -O3" CACHE STRING "")
+  SET (BUILD_HOMME_PREQX_KOKKOS TRUE CACHE BOOL "")
+  SET (ENABLE_PREQX_KOKKOS_BFB_TESTS TRUE CACHE BOOL "")
+  SET (HOMME_TESTING_PROFILE "short" CACHE STRING "")
+
+#or just these? was used before
```

Comment:
ENABLE_OPENMP is on by default, so SetCompilerFlags _should_ be finding openmp (which should add the `-qopenmp` flag). So having the flag here might be not necessary (not sure if one may even "not want it").

That said, I don't see any specification of Kokkos, so unless one passes Kokkos directives to cmake or a path to a pre-installed kokkos, the `-qopenmp` won't matter for the xx target (Kokkos will be built in serial only).

As for the traceback, it only helps in case of crash, to figure out the offending line, if I understand correctly. The cost is a small executable size increase, I think. I guess we can probably keep it.